### DISCUSSION TITLE
Hotkey for `enable_tab_bar`

### DISCUSSION THREAD

**Discussion by simonmandlik** [wezterm/wezterm#4185] [Q&A] [answered]
Would it be possible to toggle `enable_tab_bar` with a hotkey? I couldn't find any action for that, or any way to assign hotkeys to arbitrary lua code, where I would toggle the value.

**Comment by bew** [comment, marked as answer]
You can assign hotkey to arbitrary Lua code using `wezterm.action_callback` (which is a wrapper of event registration and emit)
See: https://wezfurlong.org/wezterm/config/lua/wezterm/action_callback.html

Then you can use config override to make a toggle:
See: https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html

**Reply by bew** [reply]
Here is an example of toggle (unlike @nekowinston I'm not using a global variable since we can see if there is an override already).

https://github.com/bew/dotfiles/blob/dacd162c24edec9ddc2b02b80162dcdf4d7e5cf0/gui/wezterm/cfg_keys.lua#L112-L120

**Comment by nekowinston** [comment]
See this snippet from [my dotfiles](https://github.com/nekowinston/dotfiles/blob/42d193d715cce7dabbca059d29a1360c354d7238/home/apps/wezterm/keys.lua#L17-L25)
```lua
if wezterm.GLOBAL.enable_tab_bar == nil then
  wezterm.GLOBAL.enable_tab_bar = true
end
local toggleTabBar = wezterm.action_callback(function(window)
  wezterm.GLOBAL.enable_tab_bar = not wezterm.GLOBAL.enable_tab_bar
  window:set_config_overrides({
    enable_tab_bar = wezterm.GLOBAL.enable_tab_bar,
  })
end)
```

**Comment by simonmandlik** [comment]
Thank you both!
